spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisStreamMessageHandler.java

```diff
@@ -0,0 +1,90 @@
+package org.springframework.integration.redis.outbound;
+
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.ObjectRecord;
+import org.springframework.data.redis.connection.stream.StreamRecords;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.StreamOperations;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * Take a message from a Channel and publish it to the given Redis Stream
+ */
+public class RedisStreamMessageHandler extends AbstractMessageHandler {
+
+	private final StreamOperations streamOperations;
+
+	private volatile RedisSerializer<?> serializer = StringRedisSerializer.UTF_8;
+
+	private volatile Expression streamKeyExpression;
+
+	private volatile boolean extractPayload = true;
+
+	public RedisStreamMessageHandler( String streamKey, RedisConnectionFactory connectionFactory ) {
+		Assert.notNull( streamKey, "'streamName' must not be null" );
+		Assert.notNull( connectionFactory, "'connectionFactory' must not be null" );
```

Comment:
We don't need this because `template.afterPropertiesSet();` ensures such an assertion.